capstone/src/main/java/com/google/univiz/scorecard/CollegeDataApiImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.google.univiz.scorecard;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.univiz.CollegeData;
+import com.google.univiz.CollegeDataApi;
+import com.google.univiz.api.CollegeId;
+import com.ryanharter.auto.value.gson.GenerateTypeAdapter;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+
+public class CollegeDataApiImpl implements CollegeDataApi {
+  private final ScorecardConverter scorecardConverter;
+  private final CollegeIdReaderProvider readerProvider;
+
+  @Inject
+  public CollegeDataApiImpl(
+      ScorecardConverter scorecardConverter, CollegeIdReaderProvider readerProvider) {
+    this.scorecardConverter = scorecardConverter;
+    this.readerProvider = readerProvider;
+  }
+
+  @Override
+  public List<CollegeData> getCollegesById(List<CollegeId> ids) {
+    List<CollegeData> colleges = new ArrayList<>();
+    for (CollegeId id : ids) {
+      // Then deserialize into a ScorecardResponse
+      Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+      ScorecardResponse scorecardResponse =
+          gson.fromJson(readerProvider.getReaderFromCollegeId(id), ScorecardResponse.class);
+      ScorecardData scorecard = scorecardResponse.scorecardData().get(0);
+      // Then convert to a CollegeData
+      CollegeData college = scorecardConverter.convert(scorecard);
```

Comment:
This seems like a great candidate for some streams - but we can chat about that when we get there...

Reply:
Yea I was thinking about using streams for this, but I'll adjust the loop to do only one REST call first and address the other things.